src/handlers/fix-request-body.ts

```diff
@@ -28,8 +29,26 @@ export function fixRequestBody<TReq extends BodyParserLikeRequest = BodyParserLi
   }
 
   const writeBody = (bodyData: string) => {
-    proxyReq.setHeader('Content-Length', Buffer.byteLength(bodyData));
-    proxyReq.write(bodyData);
+    let proxyData: string | Buffer = bodyData;
+
+    const contentEncoding = String(proxyReq.getHeader('Content-Encoding')).toLowerCase();
+    switch (contentEncoding) {
+      case 'br':
+        proxyData = zlib.brotliCompressSync(proxyData);
+        break;
+      case 'deflate':
+        proxyData = zlib.deflateSync(proxyData);
+        break;
+      case 'gzip':
+        proxyData = zlib.gzipSync(proxyData);
+        break;
+      case 'zstd':
+        proxyData = zlib.zstdCompressSync(proxyData);
```

Comment:
`zstd` will throw in older Node versions:

Added in: v23.8.0, v22.15.0

https://nodejs.org/api/zlib.html#zlibzstdcompresssyncbuffer-options